packages/dds/map/src/directory.ts

```diff
@@ -2275,22 +2472,43 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 					}
 				};
 				const subDirectory = this._subdirectories.get(op.subdirName);
+				// Clear the creation tracker record
+				// this.localCreationSeqTracker.delete(op.subdirName);
+				this.ackedCreationSeqTracker.delete(op.subdirName);
 				resetSubDirectoryTree(subDirectory);
 			}
 			if (op.type === "createSubDirectory") {
 				const dir = this._subdirectories.get(op.subdirName);
 				// Child sub directory create seq number can't be lower than the parent subdirectory.
 				// The sequence number for multiple ops can be the same when multiple createSubDirectory occurs with grouped batching enabled, thus <= and not just <.
-				if (this.sequenceNumber !== -1 && this.sequenceNumber <= msg.sequenceNumber) {
-					if (dir?.sequenceNumber === -1) {
-						// Only set the seq on the first message, could be more
-						dir.sequenceNumber = msg.sequenceNumber;
+				if (
+					this.seqNumCollection.seq !== -1 &&
+					this.seqNumCollection.seq <= msg.sequenceNumber
+				) {
+					if (dir?.seqNumCollection.seq === -1) {
+						// Only set the sequence collection based on the first message
+						dir.seqNumCollection.seq = msg.sequenceNumber;
+						dir.seqNumCollection.clientSeq = msg.clientSequenceNumber;
+
+						// set the creation seq in tracker
+						if (
+							!this.ackedCreationSeqTracker.has(op.subdirName) &&
+							!this.pendingDeleteSubDirectoriesTracker.has(op.subdirName)
+						) {
+							this.ackedCreationSeqTracker.set(op.subdirName, {
+								seq: msg.sequenceNumber,
+								clientSeq: msg.clientSequenceNumber,
+							});
+							if (local) {
+								this.localCreationSeqTracker.delete(op.subdirName);
```

Comment:
If the subdirectory has been created (and deleted) multiple times locally, is this tracked properly? Maybe isMessageForCurrentInstanceOfSubDirectory already handles this.

Reply:
Yes the tracking of local repeat operations is handled appropriately. It specifically monitors the initial local creation (by verifying if the created dir exists before) and the initial acked creation (by checking if the 'seq' of the directory is -1).